draft-ietf-quic-qpack.md

```diff
@@ -485,21 +513,30 @@ server's header blocks and table updates.
 
 ### Table State Synchronize
 
-After processing a set of instructions on the encoder stream, the decoder will
-emit a Table State Synchronize instruction on the decoder stream.  The
-instruction begins with the '1' one-bit pattern. The instruction specifies the
-total number of dynamic table inserts and duplications since the last Table
-State Synchronize, encoded as a 7-bit prefix integer.  The encoder uses this
-value to determine which table entries are vulnerable to head-of-line blocking.
-A decoder MAY coalesce multiple synchronization updates into a single update.
+The Table State Synchronize instruction begins with the '10' two-bit pattern.
+The instruction specifies the total number of dynamic table inserts and
+duplications since the last Table State Synchronize or Header Acknowledgement
+that increased the Largest Known Received dynamic table entry.  This is encoded
+as a 6-bit prefix integer. The encoder uses this value to determine which table
+entries might cause a stream to become blocked, as described in
+{{state-synchronization}}.
 
 ~~~~~~~~~~ drawing
   0   1   2   3   4   5   6   7
 +---+---+---+---+---+---+---+---+
-| 1 |     Insert Count (7+)     |
-+---+---------------------------+
+| 1 | 0 |   Insert Count (6+)   |
++---+---+-----------------------+
 ~~~~~~~~~~
-{:#fig-size-sync title="Table Size Synchronize"}
+{:#fig-size-sync title="Table State Synchronize"}
+
+A decoder SHOULD emit a Table State Synchronize after receiving new dynamic
+table entries if the most recently inserted entry was not the Largest Reference
```

Comment:
Since we are going to remove length prefix, 'entries' here is vague.  How many entries?  I think in the  common case the inserts will get processed before the references, but a blocking decoder would not want to send TSS without delaying to check for a referencing header block, so perhaps the SHOULD here is misleading?

Maybe it would help to describe the tradeoff -

A decoder chooses when to emit TSS instructions.  Emitting a TSS after adding each new dynamic table entry will provide the most timely feedback to the encoder, but could be redundant with other decoder feedback.  By delaying a TSS, a decoder might be able to coalesce multiple TSS instructions, or replace them entirely with Header Acknowledgements.  However, delaying too long may lead to compression inefficiencies if the encoder waits for an entry to be acknowledged before using it.